packages/xen-api/src/index.js

```diff
@@ -682,15 +682,24 @@ export class Xapi extends EventEmitter {
       // values?
       const params = args[0] === this._sessionId ? args.slice(1) : args
 
+      const isLoginWithPasswordMethod = method === 'session.login_with_password'
       error.call = {
         method,
         params:
           // it pass server's credentials as param
-          method === 'session.login_with_password'
+          isLoginWithPasswordMethod
             ? '* obfuscated *'
             : replaceSensitiveValues(params, '* obfuscated *'),
       }
 
+      if (
+        isLoginWithPasswordMethod &&
+        error.message === 'Invalid XML-RPC message'
+      ) {
+        error.message =
+          'Unexpected response, are you sure that the address is correct?'
+      }
+
```

Comment:
I don't think we should touch `xen-api`, IMHO, a special handling in `xo-web` would be more appropriate.